src/components/directory-content.js

```diff
@@ -712,13 +721,16 @@ const DirectoryContent = () => {
                         });
                     })
                     .finally(() => {
-                        setChildrenMetadata(metadata);
-                        setIsMetadataLoading(false);
+                        // discarding request for older directory
+                        if (currentChildrenRef.current === currentChildren) {
```

Comment:
This test has no effect, you just add `currentChildrenRef.current = currentChildren;` earlier, then there is no way this test could fail.

Reply:
nope, the function is asynchronous, this is a check to prevent setting erroneous data, you can test by adding a console.log in an else (throttle your network)